components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOFacade.java

```diff
@@ -95,74 +136,151 @@ public Webhook getWebhook(String webhookId, int tenantId) throws WebhookMgtExcep
+    public List<Subscription> getWebhookEvents(String webhookId, int tenantId) throws WebhookMgtException {
 
         return webhookManagementDAO.getWebhookEvents(webhookId, tenantId);
     }
 
-    @Override
-    public void updateWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
-
-        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
-        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
-        EventSubscriberService subscriberService = getSubscriberService();
-
-        withTopicRegistrationIfActive(jdbcTemplate, webhook, tenantDomain, tenantId,
-                ErrorMessage.ERROR_CODE_WEBHOOK_ADD_ERROR);
-
-        Webhook existingWebhook =
-                getWebhookWithDecryptedSecretValue(webhookManagementDAO.getWebhook(webhook.getUuid(), tenantId));
-        handleWebhookUpdate(subscriberService, webhook, existingWebhook, tenantDomain);
-
-        runTransaction(jdbcTemplate,
-                () -> webhookManagementDAO.updateWebhook(encryptAddingWebhookSecrets(webhook), tenantId),
-                ErrorMessage.ERROR_CODE_WEBHOOK_UPDATE_ERROR,
-                "updating webhook: " + webhook.getUuid() + " in tenant ID: " + tenantId, () -> {
-                    if (WebhookStatus.ACTIVE.equals(webhook.getStatus())) {
-                        unsubscribeWebhook(subscriberService, webhook, tenantDomain, webhook.getEventsSubscribed());
-                    }
-                });
-    }
-
     @Override
     public void deleteWebhook(String webhookId, int tenantId) throws WebhookMgtException {
 
         NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
-        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
-        EventSubscriberService subscriberService = getSubscriberService();
 
         runTransaction(jdbcTemplate, () -> {
                     Webhook existingWebhook = webhookManagementDAO.getWebhook(webhookId, tenantId);
-                    try {
-                        unsubscribeWebhook(subscriberService, existingWebhook, tenantDomain,
-                                existingWebhook.getEventsSubscribed());
-                    } catch (WebhookMgtException e) {
-                        LOG.warn("Error unsubscribing webhook during deletion: " + existingWebhook.getUuid(), e);
-                    }
                     webhookManagementDAO.deleteWebhook(webhookId, tenantId);
                     deleteWebhookSecrets(existingWebhook);
                 }, ErrorMessage.ERROR_CODE_WEBHOOK_DELETE_ERROR,
-                "deleting webhook: " + webhookId + " in tenant ID: " + tenantId, null);
+                "deleting webhook: " + webhookId + " in tenant ID: " + tenantId);
     }
 
     @Override
     public boolean isWebhookEndpointExists(String endpoint, int tenantId) throws WebhookMgtException {
 
+
```

Comment:
looks like this line is not necessary

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/7dde1e4229dd3a723dc8553ff68cc37345bfb1fe